test/transpose.js

```diff
@@ -0,0 +1,18 @@
+var R = require('..');
+var eq = require('./shared/eq');
+
+describe('transpose', function() {
+  it('returns an array of two arrays', function() {
+    var input = [['a', 1], ['b', 2], ['c', 3]];
+    eq(R.transpose(input), [['a', 'b', 'c'], [1, 2, 3]]);
+  });
+  it('skips elements when rows are shorter', function() {
+    var actual = R.transpose([[10, 11], [20], [], [30, 31, 32]]);
+    var expected = [[10, 20, 30], [11, 31], [32]];
+    eq(actual, expected);
+  });
```

Comment:
it seems like, say you were rendering such a table, and you wanted to transpose rows and columns, those `undefined`s are necessary information. However if you padded your initial rows with `undefined` so that they are all the same length, you would get a lossless transformation.
ie: 
```javascript
transpose([[10, 11, undefined], [20, undefined, undefined], [undefined,undefined,undefined], [30, 31, 32]]) //=>
[[10,20,30], [11, undefined, 31], [undefined, undefined, 32]]
```